test/studies/adventOfCode/2022/day12/jeremiah/day12b-par.chpl

```diff
@@ -17,7 +17,7 @@ proc readElevations() {
         start, end = (0, 0);
 
     for (row, i) in zip(elevBytes.split(b"\n"), 0..) {
-        foreach (byte, j) in zip(row, 0..) {
+        foreach (byte, j) in zip(row, 0..) with (ref start, ref end, ref elevs) {
```

Comment:
@jeremiah-corrado -- I am not familiar with the implementation here, but is this loop really order-independent? Writes to `start`, `end` and `elevs` could happen in any order, and potentially in parallel. In other words, would using `forall` be safe here ignoring performance etc?

@stonea -- I think we backed away from requiring explicit intents for:

```chpl
var Arr: [1..10] int;
foreall i in Arr.domain with (ref Arr) {
  Arr[i] = i;
}
```

Here, do we really need `ref elevs` explicitly?

Reply:
> Here, do we really need ref elevs explicitly?

I just tried taking it out and things still work. So no, it looks like `elevs` is defaulting to `ref` intent.

Reply:
> In other words, would using forall be safe here ignoring performance etc?

Yeah, the loop is order independent since there is only one "start" and "end" in the input data, and elevs is only written to once for each `(i, j)` pair.